mediapipe/graphs/instantmotiontracking/calculators/matrices_manager.cc

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <cmath>
+#include "Eigen/Dense"
+#include "Eigen/src/Core/util/Constants.h"
+#include "Eigen/src/Geometry/Quaternion.h"
+#include "absl/memory/memory.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_join.h"
+#include "mediapipe/framework/calculator_framework.h"
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/framework/port/status.h"
+#include "mediapipe/graphs/object_detection_3d/calculators/box.h"
+#include "mediapipe/graphs/object_detection_3d/calculators/model_matrix.pb.h"
+#include "mediapipe/graphs/instantmotiontracking/calculators/transformations.h"
+
+namespace mediapipe {
+
+using Matrix4fRM = Eigen::Matrix<float, 4, 4, Eigen::RowMajor>;
+
+constexpr char kFinalTranslationsTag[] = "TRANSLATION_DATA";
+constexpr char kIMUDataTag[] = "IMU_DATA";
+constexpr char kUserRotationsTag[] = "USER_ROTATIONS";
+constexpr char kUserScalingsTag[] = "USER_SCALINGS";
+constexpr char kModelMatricesTag[] = "MODEL_MATRICES";
+
+
+// Intermediary for rotation and translation data to model matrix usable by
+// gl_animation_overlay_calculator
+//
+// Input:
+//  IMU_DATA - float[3] of [roll, pitch, yaw] of device
+//  USER_ROTATIONS - UserRotations with corresponding radians of rotation
+//  TRANSLATION_DATA - All stickers final translation data (vector of
+//  Translation objects)
+// Output:
+//  MODEL_MATRICES - TimedModelMatrixProtoList of all objects to render
+//
+// Example config:
+// node{
+//  calculator: "MatricesManagerCalculator"
+//  input_stream: "IMU_DATA:imu_data"
+//  input_stream: "USER_ROTATIONS:user_rotation_data"
+//  input_stream: "TRANSLATION_DATA:final_translation_data"
+//  output_stream: "MODEL_MATRICES:model_matrices"
+// }
+
+class MatricesManagerCalculator : public CalculatorBase {
```

Comment:
nit: let's call the whole class matrices_manager_calculator, to match (and also stay consistent with our other calculators)